plutus-core/plutus-core/src/PlutusCore/Core/Instance/Pretty/Classic.hs

```diff
@@ -15,61 +16,88 @@ module PlutusCore.Core.Instance.Pretty.Classic () where
 import           PlutusPrelude
 
 import           PlutusCore.Core.Instance.Pretty.Common ()
-import           PlutusCore.Core.Instance.Recursive
 import           PlutusCore.Core.Type
 import           PlutusCore.Pretty.Classic
 import           PlutusCore.Pretty.PrettyConst
 
-import           Data.Functor.Foldable
 import           Data.Text.Prettyprint.Doc
 import           Data.Text.Prettyprint.Doc.Custom
 import           Universe
 
-instance PrettyBy (PrettyConfigClassic configName) (Kind a) where
-    prettyBy _ = cata a where
-        a TypeF{}             = "(type)"
-        a (KindArrowF _ k k') = parens ("fun" <+> k <+> k')
-
-instance (PrettyClassicBy configName tyname, GShow uni) =>
-        PrettyBy (PrettyConfigClassic configName) (Type tyname uni a) where
-    prettyBy config = cata a where
-        a (TyAppF _ t t')     = brackets (t <+> t')
-        a (TyVarF _ n)        = prettyName n
-        a (TyFunF _ t t')     = parens ("fun" <+> t <+> t')
-        a (TyIFixF _ pat arg) = parens ("ifix" <+> pat <+> arg)
-        a (TyForallF _ n k t) = parens ("all" <+> prettyName n <+> prettyBy config k <+> t)
-        a (TyBuiltinF _ n)    = parens ("con" <+> pretty n)
-        a (TyLamF _ n k t)    = parens ("lam" <+> prettyName n <+> prettyBy config k <+> t)
-
-        prettyName = prettyBy config
+instance Pretty ann => PrettyBy (PrettyConfigClassic configName) (Kind ann) where
+    prettyBy config = \case
+        Type ann           ->
+            parens (vsep' (consAnnIf config ann
+                ["type"]))
+        KindArrow ann k k' ->
+            parens ("fun" </> vsep' (consAnnIf config ann
+                [prettyBy config k, prettyBy config k']))
```

Comment:
I had to change all the classic prettyprinters to teach them how to print annotations on demand.